app/connectors/GoodsRecordConnector.scala

```diff
@@ -32,14 +34,35 @@ class GoodsRecordConnector @Inject() (config: Configuration, httpClient: HttpCli
 ) {
   private val tgpRouterBaseUrl: Service = config.get[Service]("microservice.services.trader-goods-profiles-router")
   private val goodsRecordUrl            = url"$tgpRouterBaseUrl/trader-goods-profiles-router/records"
+  private val clientIdHeader            = ("X-Client-ID", "tgp-frontend")
+
+  private def getRecordUrl(eori: String, recordId: String) =
+    url"$tgpRouterBaseUrl/trader-goods-profiles-router/$eori/records/$recordId"
 
   def submitGoodsRecord(goodsRecord: GoodsRecord)(implicit
     hc: HeaderCarrier
   ): Future[CreateGoodsRecordResponse] =
     httpClient
       .post(goodsRecordUrl)
-      .setHeader(header = ("X-Client-ID", "tgp-frontend"))
+      .setHeader(clientIdHeader)
       .withBody(Json.toJson(CreateRecordRequest.map(goodsRecord)))
       .execute[HttpResponse]
       .map(response => response.json.as[CreateGoodsRecordResponse])
+
+  def getRecord(eori: String, recordId: String)(implicit
+    hc: HeaderCarrier
+  ): Future[GetGoodsRecordResponse] =
+    httpClient
+      .get(getRecordUrl(eori, recordId))
+      .addHeaders(clientIdHeader)
+      .execute[HttpResponse]
+      .flatMap { response =>
+        response.status match {
+          case OK =>
```

Comment:
what happens when we don't get an OK? Will it be throwing an exception as it won't be a successful response? If so we can just parse the json in the flatmate and not worry about a match

you get the code below by default if you use response.json.as[GetGoodsRecordResponse]

.map(result => Future.successful(result))
               .recoverTotal(error => Future.failed(JsResult.Exception(error)))

Reply:
Good point, I think that would work, it should fail the future with an exception in cases where things break anyway :)
I'll try it out

Reply:
Did something similarly simple now :)